developer-docs/bolt-server.md

```diff
@@ -229,52 +224,62 @@ curl -v --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT
+bolt server during development at `scripts/server_client.rb`. This server
 expects to use the puppet-server container and target nodes from bolts spec
-environment so run `docker-compose up -d --build` from the spec directory before
-trying to use it.
+environment so run `docker-compose up -d --build` from the spec directory
+before trying to use it.
 
 ```
-bundle exec scripts/server_client.rb sample::echo ssh '{"message": "hey"}'
+bundle exec scripts/server_client.rb sample::echo <TARGET> '{"message": "hey"}'
+```
+Where `<TARGET>` is either:
+* A vmpooler VM. To use this, replace `<TARGET>` above with the hostname.
+* One of the containers brought up by the `docker-compose` in the `spec` directory. To use these, you'll want to get the IP of the container bolt-server is running on:
+```
+docker exec -it bolt_boltserver_1 /sbin/ip route|awk '/default/ { print $3 }'
+# Should return an IP such as 172.20.0.1
+```
+Then append the port of one of the 3 containers to that IP: `20022` (for an ubuntu node with no agent), `20023` (for a puppet 5 agent), or `20024` (for a puppet 6 agent). It's also helpful to include the protocol (`ssh`), user (`bolt`), and password (`bolt`) in the URI.
+
+So your request will be something like:
+```
+bundle exec scripts/server_client.rb sample::echo ssh://bolt:bolt@172.20.0.1:20022 '{"message": "hey"}'
 ```
 
 ### With cURL
 The following is an example request body update it and save it to request.json
 
 ```
-{
-  "target": {
-    "hostname": [hostname of target],
-    "user": "root",
-    "private-key-content": [Contents of ssh private key as a string],
-    "host-key-check": false
-  },
-  "task": {
-    "name": "echo",
-    "metadata": {
-      "description": "Echo a message",
-      "parameters": {
-        "message": "Default string"
-      }
-    },
-    "file": {
-      "filename": "echo.sh",
-      "file_content": "IyEvdXNyL2Jpbi9lbnYgYmFzaAplY2hvICRQVF9tZXNzYWdlCg==\n"
-    }
-  },
-  "parameters": {
-    "message": "Hello world"
-  }
-}
+{"task":{
+  "metadata":{},
+  "name":"sample::echo",
+  "files":[{
+    "filename":"echo.sh",
+    "sha256":"c5abefbdecee006bd65ef6f625e73f0ebdd1ef3f1b8802f22a1b9644a516ce40",
+    "size_bytes":64,
+    "uri":{
+      "path":"/puppet/v3/file_content/tasks/sample/echo.sh",
+      "params":{
+        "environment":"production"}}}]},
+"target":{
+  "hostname":"172.20.0.1",
+  "user":"bolt",
+  "password":"bolt",
+  "port": 20022,
+  "host-key-check":false},
+"parameters":{
+  "message":"Hello World"}}
 ```
-You should then be able to post it with:
+**Verify that the target information** is correct, and change it if you want to use a different target. You can find other example requests in the `examples` directory.
 
+You should then be able to post it with:
 ```
 curl -X POST -H "Content-Type: application/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT $BOLT_ROOT/ssh/run_task
```

Comment:
Extra `$BOLT_ROOT` there. Should be `curl -X POST -H ation/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY  $BOLT_ROOT/ssh/run_task`. Also i think it is important to show the full output 
```
{"node":"172.18.0.1","status":"success","result":{"_output":"ac80223bd3b4 got passed the message: hey\n"}}
```
The `node` value might be confusing bc it is not running on the docker host (your laptop) instead it is running against the bolt-spec container. Showing the `_output` is helpful as it should be the container ID the task is run against
```
ac80223bd3b4        bolt-spec           "/usr/sbin/sshd -D"      30 minutes ago      Up 30 minutes               0.0.0.0:20022->22/tcp      ubuntu_node
```